haystack_experimental/dataclasses/breakpoints.py

```diff
@@ -0,0 +1,93 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from dataclasses import dataclass
+from typing import Optional, Set, Union
+
+
+@dataclass
+class Breakpoint:
+    """
+    A dataclass to hold a breakpoint for a component.
+    """
+
+    component_name: str
+    visit_count: int
+
+    def __hash__(self):
+        return hash((self.component_name, self.visit_count))
+
+    def __eq__(self, other):
+        if not isinstance(other, Breakpoint):
+            return False
+        return self.component_name == other.component_name and self.visit_count == other.visit_count
+
+    def __str__(self):
+        return f"Breakpoint(component_name={self.component_name}, visit_count={self.visit_count})"
+
+    def __repr__(self):
+        return self.__str__()
+
+
+@dataclass
+class ToolBreakpoint(Breakpoint):
+    """
+    A dataclass to hold a breakpoint that can be used to debug a Tool.
+
+    If tool_name is None, it means that the breakpoint is for every tool in the component.
+    Otherwise, it means that the breakpoint is for the tool with the given name.
+    """
+
+    tool_name: Optional[str] = None
```

Comment:
probably here we want a `Optional[List[str]]` to set breakpoints at multiple tools